lua/telescope/builtin.lua

```diff
@@ -45,14 +46,18 @@ builtin.git_files = function(opts)
     opts.cwd = vim.fn.expand(opts.cwd)
   end
 
+  opts.track_results = vim.g["telescope_cache_results"]
```

Comment:
It seems `track_results` is related to both the sorter and the finder, it should probably not be passed in by the user.

I'd say it should be determined by some interaction of the finder and the sorter.